jsettlers.graphics/src/main/java/jsettlers/graphics/map/controls/original/panel/content/settlers/profession/ProfessionPanel.java

```diff
@@ -1,66 +1,245 @@
+	public enum EProfessionType {
+		CARRIER("Carriers", true, EMovableType.BEARER),
+		DIGGER("Diggers", false, EMovableType.DIGGER),
+		BUILDER("Builders", false, EMovableType.BRICKLAYER);
+
+		public final String label;
+		public final boolean min;
+		public final EMovableType moveableType;
+
+		private EProfessionType(String label, boolean min, EMovableType movableType) {
+			this.label = label;
+			this.min = min;
+			this.moveableType = movableType;
+		}
+	}
+
+	public class ContentPanel extends Panel {
+		private ShortPoint2D position;
+		private final SettlerPanel diggerPanel;
+		private final SettlerPanel carrierPanel;
+		private final SettlerPanel builderPanel;
+
+		public ContentPanel() {
+			super(118f, 216f);
+			this.position = new ShortPoint2D(0, 0);
+			this.carrierPanel = new SettlerPanel(widthInPx, EProfessionType.CARRIER);
+			this.diggerPanel = new SettlerPanel(widthInPx, EProfessionType.DIGGER);
+			this.builderPanel = new SettlerPanel(widthInPx, EProfessionType.BUILDER);
+
+			add(Panel.box(new Label(Labels.getString("Settlers"), EFontSize.NORMAL), widthInPx, 20f), 0f, 0f);
+			add(carrierPanel, 0f, 20f);
+			add(diggerPanel, 0f, 50f);
+			add(builderPanel, 0f, 80f);
+		}
+
+		public void setup(IProfessionSettings settings) {
+			this.carrierPanel.setRatio(settings.getMinBearerRatio());
+			this.carrierPanel.setCurrentRatio(settings.getCurrentBearerRatio());
+
+			this.diggerPanel.setRatio(settings.getMaxDiggerRatio());
+			this.diggerPanel.setCurrentRatio(settings.getCurrentDiggerRatio());
+
+			this.builderPanel.setRatio(settings.getMaxBricklayerRatio());
+			this.builderPanel.setCurrentRatio(settings.getCurrentBricklayerRatio());
+
+			update();
+		}
+
+		public float getTotalRatio() {
+			return carrierPanel.ratio + diggerPanel.ratio + builderPanel.ratio;
+		}
+
+		public void setPosition(ShortPoint2D position) {
+			this.position = position;
+		}
+
+		public class SettlerPanel extends Panel {
+			private float ratio;
+			private float currentRatio;
+			private final Label label;
+			private final EProfessionType type;
+
+			public SettlerPanel(float width, EProfessionType type) {
+				super(width, 30f);
+				this.ratio = 0f;
+				this.currentRatio = 0f;
+				this.label = new Label("...", EFontSize.NORMAL, EHorizontalAlignment.LEFT);
+				this.type = type;
+
+				add(new UpDownArrows(type), 10f, 5f);
+				add(Panel.box(label, width, 20f), 28f, 5f);
+			}
+
+			@Override
+			public void update() {
+				super.update();
+				this.label.setText(MessageFormat.format("{0} {1} {2} ({3})", (this.type.min ? '>' : '<'), formatPercentage(ratio), Labels.getString(this.type.label), formatPercentage(currentRatio)));
+			}
+
+			private String formatPercentage(float value) {
+				return (int)(value * 100f) + "%";
+			}
+
+			public void setRatio(float ratio) {
+				this.ratio = ratio;
+			}
+
+			public void setCurrentRatio(float currentRatio) {
+				this.currentRatio = currentRatio;
+			}
+
+			public class UpDownArrows extends Panel {
```

Comment:
This already exists as jsettlers.graphics.ui.CountArrows and does not have the latency issue.

Reply:
Ok i reused the existing one.